tool/tsh/common/access_request.go

```diff
@@ -449,26 +480,63 @@ func searchRequestableRoles(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
-	if len(allRoles) == 0 {
-		fmt.Fprintln(cf.Stdout(), "No requestable roles found.")
-		return nil
-	}
-
-	columns := []string{"Role", "Description"}
-	var rows [][]string
+	rows := make([]requestableRoleRow, 0, len(allRoles))
 	for _, r := range allRoles {
-		rows = append(rows, []string{r.Name, r.Description})
+		rows = append(rows, requestableRoleRow{
+			Role:        r.Name,
+			Description: r.Description,
+		})
 	}
 
-	var table asciitable.Table
-	if cf.Verbose {
-		table = asciitable.MakeTable(columns, rows...)
-	} else {
-		table = asciitable.MakeTableWithTruncatedColumn(columns, rows, "Description")
-	}
+	return printRequestableRoles(cf, rows)
+}
 
-	_, err = table.AsBuffer().WriteTo(cf.Stdout())
-	return trace.Wrap(err)
+func printRequestableRoles(cf *CLIConf, rows []requestableRoleRow) error {
+	format := strings.ToLower(cf.Format)
+
+	switch format {
+	case teleport.Text, "":
+		if len(rows) == 0 {
+			fmt.Fprintln(cf.Stdout(), "No requestable roles found.")
+			return nil
+		}
+
+		columns, rows, err := asciitable.MakeAsciitableColumnsAndRows(rows, nil)
+		if err != nil {
+			return err
+		}
+
+		var table asciitable.Table
+		if cf.Verbose {
+			table = asciitable.MakeTable(columns, rows...)
+		} else {
+			table = asciitable.MakeTableWithTruncatedColumn(columns, rows, "Description")
+		}
+
+		if _, err := table.AsBuffer().WriteTo(cf.Stdout()); err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+
+	case teleport.JSON, teleport.YAML:
+		var out []byte
+		var err error
+
+		if format == teleport.JSON {
+			out, err = utils.FastMarshalIndent(rows, "", "  ")
```

Comment:
Should we use `utils.WriteJSONArray` here instead? I think that ensures we write an empty `[]` if there was no data.

(Similarly, we have `utils.WriteYAML` for the same purpose)